### PR TITLE
Added compatability to current version of carsus atomic data

### PR DESCRIPTION
**Pull request by Rodot-**
Changed the way that the atomic data objects are instantiated in TARDIS to handle atomic data produced by carsus

**Motivation and context**
Currently, carsus and tardis are incompatible when using atomic data containing collisional data.

**How has this been tested?**
Local tests with current carsus branch

- [x] Testing pipeline
- [ ] Other

**Examples**

**Type of change**

- [x] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] None of the above 

**Checklist**

- [ ] I have updated the documentation according to my changes
- [ ] I have built the documentation by applying the `build_docs` label to this pull request (if you don't have enough privileges a reviewer will do it for you)
- [x] I have requested two reviewers for this pull request

### CODE REVIEW

**Comment by Rodot-** [comment]
Should be noted, this branch now seems to work but it must be run single-threaded.

**Comment by Rodot-** [comment]
We should start thinking about writing tests, but that could go into a later PR, what does everyone think?

**Comment by epassaro** [comment]
> We should start thinking about writing tests, but that could go into a later PR, what does everyone think?

We are currently adding more metadata and versioning Carsus format, so the key checkingássigning must be handled differently. I hope we have this ready for Thursday.

File: tardis/io/atom_data/base.py [hunk 1]
```diff
@@ -165,25 +165,70 @@ def from_hdf(cls, fname=None):
...
                     dataframes[name] = store[name]
                 except KeyError:
                     logger.debug(f"Dataframe does not contain {name} column")
                     nonavailable.append(name)
 
+            if "metadata" in store:
+                carsus_version = (
+                    store["metadata"].loc[("format", "version")].value
+                )
+                if carsus_version == "1.0":
```
[the remark this answers is not in the document]
Comment (Rodot-) [reply tardis/io/atom_data/base.py:180]: version numbers aren't general floats.  Typically they would be tuples of ints but this is the formal we decided on for carsus.

File: tardis/plasma/properties/continuum_processes.py [hunk 2]
```diff
@@ -603,7 +603,9 @@ def calculate(
         )
         num_neg_elements = (gamma_corr < 0).sum().sum()
         if num_neg_elements:
-            raise PlasmaException("Negative values in CorrPhotoIonRateCoeff.")
+            raise PlasmaException(
+                "Negative values in CorrPhotoIonRateCoeff.  Try raising the number of montecarlo packets."
+            )
```
[the remark this answers is not in the document]
Comment (Rodot-) [reply tardis/plasma/properties/continuum_processes.py:608]: Not really, it's a packet statistics thing.  If any of the estimators get too few hits in a shell then the resulting values won't end up working.  In the future we'd like to be able to have more concrete solutions to this, but this seems to be how it works in other radtrans codes as well.

File: tardis/io/atom_data/base.py [hunk 3]
```diff
@@ -192,6 +237,10 @@ def from_hdf(cls, fname=None):
 
             try:
                 atom_data.version = store.root._v_attrs["database_version"]
+                if hasattr(atom_data.version, "decode"):
```
Comment (sonachitchyan) [inline tardis/io/atom_data/base.py:240]: Maybe we should get rid of this part now, as it's never encoded in the first place?

Comment (epassaro) [reply tardis/io/atom_data/base.py:240]: I'd say yes, the `version` attribute is new and was never encoded. That's different for MD5 and UUID1 that existed in previous versions.

File: tardis/io/atom_data/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rodot-) [reply tardis/io/atom_data/base.py:180]: Do we want to do this now and here?

**Review by sonachitchyan — APPROVED** [review]

**Comment by epassaro** [comment]
Is this ready? @sonachitchyan @Rodot-

**Comment by epassaro** [comment]
> **How has this been tested?** It hasn't really
> 
> *

We can say this has been tested against old Carsus atomic files.

New atomic files were tested locally, right?

**Comment by Rodot-** [comment]
Yep, we tested the new ones locally and the old ones are tested by the pipeline

File: tardis/io/atom_data/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rodot-) [reply tardis/io/atom_data/base.py]: done

**Comment by Rodot-** [comment]
@epassaro @andrewfullard @chvogl I think this is ready to merge if you add your approvals, unless anyone has any final comments.

**Review by epassaro — APPROVED** [review]

**Pull request merged**